src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -30,8 +30,10 @@ internal sealed class HttpHandlerDiagnosticListener : ListenerHandler
     {
         internal static readonly AssemblyName AssemblyName = typeof(HttpHandlerDiagnosticListener).Assembly.GetName();
         internal static readonly string ActivitySourceName = AssemblyName.Name;
+        internal static readonly string FrameworkActivitySourceName = "System.Net.Http";
         internal static readonly Version Version = AssemblyName.Version;
         internal static readonly ActivitySource ActivitySource = new(ActivitySourceName, Version.ToString());
+        internal static readonly bool IsNet7OrGreater = typeof(HttpClient).Assembly.GetName().Version.Major >= 7;
```

Comment:
Thanks @tarekgh @noahfalk - The second option could introduce an issue - Thinking about the edge case - If I have sampler always returning `ActivitySamplingResult.None`. In this case HttpClient will fall back to check if there is a diagnostic listener [present](https://github.com/dotnet/runtime/blob/b55ae3df2ea7b165a8bd7ddb9cba66c70efcf1e2/src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs#L66), if there is one it will create activity using `new Activity()`. For this case, we will never get an activity with source set.

I am checking source property as well [here](https://github.com/open-telemetry/opentelemetry-dotnet/blob/09e6a82a704e2bbd2ff63c130d4fc8b1ef91a99b/src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs#L64) as there could be another diagnosticsource listener which could ultimately cause activity to be created.

Please correct me if I am wrong.

Reply:
You could make your listener return false in response to the `s_diagnosticListener.IsEnabled(DiagnosticsHandlerLoggingStrings.ActivityName, requestMessage)` as a potential optimization, but I agree your check for the source is still needed because you aren't in control of the `Activity.Current is not null` portion of that check. I think this PR is doing the right things.